### DISCUSSION TITLE
Issue with high variance in Pmsim$new

### DISCUSSION THREAD

**Discussion by HMueller0** [LAPKB/Pmetrics#190] [Q&A] [answered]
Hello,
I am new to the package, and unfortunately I encounter a problem when running a simulation. I am adding data from a reported model with poppar and want to perform a simulation for a couple of doses with PM_sim$new.

```
> weights <- 1
> mean <- c(0.1884,0.920)
> cov <- matrix(rep(0, length(mean)**2), ncol = length(mean))  
> diag(cov)<- (c(0.277,0.522)*mean)**2 
>  poppar_test <- list(weights, mean, cov)
> 
> sim_test<-PM_sim$new(
>                      poppar=poppar_test,
>                      data=DATA1,
>                      model = MODEL,
>                      nsim=1000)
> 
```

But then I get several messages: 

> Note: The following floating-point exceptions are signalling: IEEE_INVALID_FLAG IEEE_OVERFLOW_FLAG
> Use, for example, SIMparse(“simout1.txt”) to extract simulator output for analysis or plotting.
> 
> Processing 1 simulated data file(s)
>   |                                                  |   0%Error in scan(allfiles[n], skip = amtlines[a], n = nsim * (nobs + 1),  : 
>   scan() expected 'a real', got '0.3734152+101'

If I change the input (coefficients of variation) in the covariation matrix to lower values (e.g. 0.1 instead of the 0.277 and 0.522, the error messages disappears. What can I do solve this?

Thank you very much for your help!

**Comment by mnneely** [comment, marked as answer]
There is something off about your simulation, because you are generating outputs in excess of 10^99. Try adding limits to your simulated parameter ranges to avoid simulating unphysiologic values, such as negative.  

The model building app will help you specify limits. Launch it with `build_model()`.  You can also specify limits manually in a model list or file.

In a model list, specify the limits like this:

```
pri = list(
Ke = ab(0,5),
V = ab(0,100)
)
```

In a model file, like this:

```
#PRI
Ke, 0, 5
V, 0, 100
```

Once you have the limits, then simply include `limits = NA` in your `PM_sim$new()` call. See `?SIMrun` for more help with limits. You can specify the limits manually for all the parameters in the `PM_sim$new()` call, but I find it easier to do it the way I've outlined above.

**Reply by HMueller0** [reply]
Thanks so much for your help! =)
Yes, I had defined the model with build_model() and the range did not include negatives or 0, but I did not use the argument `limits = NA `before. Now I added it and it works!

**Comment by mnneely** [comment]
Excellent!
